Proposal for #5, #6 and #7; includes README update

SIPS1980 opened the pull request:
As I did submit the issues #5, #6 and #7 I figured I also propose a solution. 

The reason behind bringing this up is because I'm using this NPM package in a VSCode extension to improve the Markdown Preview: [Markdown Preview Include Support](https://marketplace.visualstudio.com/items?itemName=stamminger.vscode-markdown-preview-include) ... will add a "Credits" note to this project as well with the next release.

Hope my proposal's help - thanks a lot in advance for consideration :-)

SIPS1980 committed:
Proposal for #5, #6 and #7; includes README update

camelaissani commented:
Thanks for your contribution.
Could you fix the PR? The changes break the tests.

About the new options you have introduced, do you really need to customize the error messages?

SIPS1980 committed:
Update Tests

SIPS1980 commented:
Updated the tests and made also some slight changes. 

To make the test work on Windows 10 I added 2 'macros' specifically to execute the test on Windows 10. Maybe there is a better way to do this - it's my first time to use 'make'

Following is the error message when I ran 'make test' on Windows 10 (in particular the slashes '/' causes the problems): 
![image](https://user-images.githubusercontent.com/25469040/59159928-6b3aeb00-8a9e-11e9-85cb-f8a1da0c3845.png)

Once i changed th slashes '/' to backslashes '\' it worked just fine:
![image](https://user-images.githubusercontent.com/25469040/59159963-c8cf3780-8a9e-11e9-9cb4-0b4414c3fa5c.png)
![image](https://user-images.githubusercontent.com/25469040/59159967-d08edc00-8a9e-11e9-936f-a2996aad8203.png)

SIPS1980 commented:
Regarding the additional options:

From a prospective of MVP (Minimum viable product) the additional options I do technically don't need them. There are 2 additional considerations though:

* If the plugin is part of generate a page for publishing it to the web, the current message which is written into the text would expose the local path (This may be a security concern) - hence the possibility to change the the respective messages
* Being able to switch between throwing an error and just writing the message of either file not found or circular reference directly into the output is more for backwards compatibility purposes. One may have already integrated markdown-it-include int the chain and relies on an error message to be thrown in order to take proper steps. Some one new (like me) may not want that - hence the default continue to be throwing a message.

Bottom line:
* I technically don't need to be able to change the message.
* I do need the option to disable throwing the error so I can use it in the VSCode extension: https://marketplace.visualstudio.com/items?itemName=stamminger.vscode-markdown-preview-include (Note: I already added this project in the credits)

camelaissani commented on index.js:
```diff
@@ -7,37 +7,55 @@ var INCLUDE_RE = /\!{3}\s*include\s*\(\s*(.+?)\s*\)\s*\!{3}/i;
     }
     while ((cap = includeRe.exec(src))) {
       filePath = path.resolve(rootdir, cap[1].trim());
+      mdSrc = '';
```
I would have used a new variable to handle the error message. Like `errorMessage`.
